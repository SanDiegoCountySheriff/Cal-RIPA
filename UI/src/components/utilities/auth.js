import { PublicClientApplication } from '@azure/msal-browser'

export const doSignIn = async () => {
  await this.$msalInstance
    .loginPopup({})
    .then(() => {
      const myAccounts = this.$msalInstance.getAllAccounts()
      this.account = myAccounts[0]
      this.$emitter.emit('login', this.account)
    })
    .catch(error => {
      console.error(`error during authentication: ${error}`)
    })
}
