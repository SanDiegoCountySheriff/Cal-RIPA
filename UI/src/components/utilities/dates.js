export const formatDate = dateStr => {
  if (dateStr && dateStr.length > 0) {
    const str = dateStr.replace(/-/g, '')
    const year = str.substring(0, 4)
    const month = str.substring(4, 6)
    const day = str.substring(6, 8)
    return `${year}-${month}-${day}`
  }

  return ''
}
