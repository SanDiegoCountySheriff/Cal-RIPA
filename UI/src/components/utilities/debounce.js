export const debounce = (func, delay, immediate) => {
  let timerId
  return (...args) => {
    const boundFunc = func.bind(this, ...args)
    clearTimeout(timerId)
    if (immediate && !timerId) {
      boundFunc()
    }
    const calleeFunc = immediate
      ? () => {
          timerId = null
        }
      : boundFunc
    timerId = setTimeout(calleeFunc, delay)
  }
}
