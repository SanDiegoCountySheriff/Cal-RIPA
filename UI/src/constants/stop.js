export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
