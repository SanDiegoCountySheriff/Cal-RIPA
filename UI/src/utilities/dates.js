import differenceInHours from 'date-fns/differenceInHours'
import { differenceInYears as diffInYears } from 'date-fns'
import format from 'date-fns/format'
import isValid from 'date-fns/isValid'
import endOfDay from 'date-fns/endOfDay'

export const formatDate = dateStr => {
  if (dateStr && dateStr.length > 0) {
    const str = dateStr.replace(/-/g, '')
    const year = str.substring(0, 4)
    const month = str.substring(4, 6)
    const day = str.substring(6, 8)
    return `${year}-${month}-${day}`
  }

  return ''
}

export const isValidDate = dateStr => {
  const date = parseDate(dateStr, '00:00')
  return isValid(date)
}

const parseDate = (dateStr, timeStr) => {
  return new Date(`${dateStr}T${timeStr}`)
}

export const dateNotInFuture = (dateStr, timeStr) => {
  const date = parseDate(dateStr, timeStr || '00:00')
  return new Date().getTime() >= date.getTime()
}

export const differenceInYears = date => {
  const diff = diffInYears(new Date(), new Date(date))
  return diff
}

export const dateWithinLastHours = (dateStr, timeStr, hours) => {
  const date = parseDate(dateStr, timeStr)
  const diff = differenceInHours(new Date(), date)
  return diff < hours
}

export const formatDateTime = (dateStr, timeStr) => {
  return new Date(`${dateStr}T${timeStr}`)
}

export const formatShortDate = dateStr => {
  const date = parseDate(dateStr, '00:00')
  return format(new Date(date), 'yyyy-MM-dd')
}

export const uniqueId = () => {
  return format(new Date(), 'yyyyMMddkkmm')
}

export const formatToIsoCurrentDate = () => {
  return format(endOfDay(new Date()), `yyyy-MM-dd'T'HH:mm:ss.SSSxxx`)
}

export const formatToIsoDate = date => {
  return format(endOfDay(date), `yyyy-MM-dd'T'HH:mm:ss.SSSxxx`)
}
