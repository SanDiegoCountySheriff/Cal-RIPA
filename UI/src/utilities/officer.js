export const getOfficerStartDate = () => {
  const startDate = localStorage.getItem('ripa_officer_start_date')
  return startDate || null
}

export const getOfficerYearsExperience = () => {
  const yearsExperience = localStorage.getItem('ripa_officer_years_experience')
  return +yearsExperience || null
}

export const getOfficerAssignment = () => {
  const assignment = localStorage.getItem('ripa_officer_assignment')
  return +assignment || null
}

export const getOfficerOtherType = () => {
  const otherType = localStorage.getItem('ripa_officer_other_type')
  return otherType || null
}
