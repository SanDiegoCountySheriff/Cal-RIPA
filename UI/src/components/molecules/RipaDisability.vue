<template>
  <div class="ripa-disability tw-pb-4">
    <ripa-form-header
      title="Perceived or Known Disability"
      required
      subtitle="ยง999.226(a)(9)"
      :on-open-statute="onOpenStatute"
    >
    </ripa-form-header>

    <v-container>
      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <ripa-switch
            v-model="model.person.anyDisabilities"
            label="Any Disabilities?"
            :disabled="disabled"
            :max-width="200"
            @input="handleInput"
          ></ripa-switch>

          <template v-if="model.person.anyDisabilities">
            <ripa-check-group
              v-model="model.person.perceivedOrKnownDisability"
              :disabled="disabled"
              :items="getDisabilityItems"
              :rules="disabilityRules"
              @input="handleInput"
            >
            </ripa-check-group>
          </template>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import RipaFormHeader from '@/components/molecules/RipaFormHeader'
import RipaFormMixin from '@/components/mixins/RipaFormMixin'
import RipaCheckGroup from '@/components/atoms/RipaCheckGroup'
import RipaSwitch from '@/components/atoms/RipaSwitch'
import { DISABILITIES } from '@/constants/form'

export default {
  name: 'ripa-disability',

  mixins: [RipaFormMixin],

  components: {
    RipaFormHeader,
    RipaCheckGroup,
    RipaSwitch,
  },

  data() {
    return {
      disabilityItems: DISABILITIES,
      viewModel: this.loadModel(this.value),
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
    },

    getDisabilityItems() {
      if (this.viewModel.person.isStudent) {
        return this.disabilityItems
      }

      return this.disabilityItems.filter(item => item.value !== 7)
    },

    disabilityRules() {
      const checked = this.viewModel.person.anyDisabilities
      const options = this.viewModel.person.perceivedOrKnownDisability
      return [
        (checked && options.length > 0) ||
          'At least one disability is required',
      ]
    },
  },

  methods: {
    handleInput() {
      this.$emit('input', this.viewModel)
    },

    updatePerceivedOrKnownDisabilityModel() {
      this.$nextTick(() => {
        this.viewModel.person.perceivedOrKnownDisability = []
      })
    },
  },

  watch: {
    value(newVal) {
      this.viewModel = this.loadModel(newVal)
    },

    'viewModel.person.anyDisabilities': {
      handler(newVal, oldVal) {
        if (oldVal !== newVal) {
          this.updatePerceivedOrKnownDisabilityModel()
        }
      },
    },
  },

  props: {
    value: {
      type: Object,
      default: () => {},
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
