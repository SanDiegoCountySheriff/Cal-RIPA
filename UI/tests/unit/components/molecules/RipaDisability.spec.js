import RipaDisability from '@/components/molecules/RipaDisability.vue'
import { shallowMount, mount } from '@vue/test-utils'
import { defaultStop } from '@/utilities/stop'
import Vuetify from 'vuetify'

describe('Ripa Disability', () => {
  let vuetify
  let wrapper
  let stop

  beforeEach(() => {
    vuetify = new Vuetify()
    stop = defaultStop()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaDisability, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaDisability, {
      vuetify,
      propsData: {
        value: stop,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
