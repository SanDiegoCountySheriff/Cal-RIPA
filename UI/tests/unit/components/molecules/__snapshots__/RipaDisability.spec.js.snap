// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Disability should match snapshot 1`] = `
<div class="tw-pb-4">
  <div class="ripa-form-header">
    <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
      <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived or Known Disability</span></div>
    </div>
  </div>
  <div class="container">
    <div class="row no-gutters">
      <div class="col-sm-12 col-12">
        <div class="ripa-switch" style="max-width: 300px;">
          <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-8" role="switch" type="checkbox" aria-disabled="false" value="">
                  <div class="v-input--selection-controls__ripple"></div>
                  <div class="v-input--switch__track theme--light"></div>
                  <div class="v-input--switch__thumb theme--light">
                    <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                  </div>
                </div><label for="input-8" class="v-label theme--light" style="left: 0px; position: relative;"><span class="main-label mr-3">
        Any Disabilities?
      </span> <span class="switch-label" style="right: -45px;">
        No
      </span></label>
              </div>
            </div>
          </div>
        </div>
        <!---->
      </div>
    </div>
  </div>
</div>
`;
